voltixApp/VoltixApp/Services/Solana/SolanaService.swift

```diff
@@ -0,0 +1,128 @@
+import Foundation
+import SwiftUI
+
+	// MARK: - SolanaService
+@MainActor
+class SolanaService: ObservableObject {
+	static let shared = SolanaService()
+	private init() {}
+	
+	@Published var transactionResult: String?
+	@Published var balance: Int?
+	@Published var recentBlockHash: String?
+	@Published var feeInLamports: String?
+	
+	func solBalanceInUSD(usdPrice: Double?, includeCurrencySymbol: Bool = true) -> String? {
+		guard let usdPrice = usdPrice,
+			  let solBalance = balance else { return nil }
+		
+		let balanceSOL = Double(solBalance) / 1_000_000_000.0
+		let balanceUSD = balanceSOL * usdPrice
+		
+		let formatter = NumberFormatter()
+		
+		if includeCurrencySymbol {
+			formatter.numberStyle = .currency
+			formatter.currencyCode = "USD"
+		} else {
+			formatter.numberStyle = .decimal
+			formatter.maximumFractionDigits = 2
+			formatter.minimumFractionDigits = 2
+			formatter.decimalSeparator = "."
+			formatter.groupingSeparator = ""
+		}
+		
+		return formatter.string(from: NSNumber(value: balanceUSD))
+	}
+	
+	var formattedSolBalance: String? {
+		
+		guard let solAmountInt = balance else {
+			return "Balance not available"
+		}
+		
+		let solAmount = Double(solAmountInt)
+		let balanceSOL = solAmount / 1_000_000_000.0 // Adjusted for SOL
+		
+		let formatter = NumberFormatter()
+		formatter.numberStyle = .decimal
+		formatter.maximumFractionDigits = 9
+		formatter.minimumFractionDigits = 0
+		formatter.groupingSeparator = ""
+		formatter.decimalSeparator = "."
+		return formatter.string(from: NSNumber(value: balanceSOL))
+	}
+	
+	private let rpcURL = URL(string: Endpoint.solanaServiceAlchemyRpc)!
+	private let jsonDecoder = JSONDecoder()
+	
+	func sendSolanaTransaction(encodedTransaction: String) async {
+		do {
+			let requestBody: [String: Any] = [
+				"jsonrpc": "2.0",
+				"id": 1,
+				"method": "sendTransaction",
+				"params": [encodedTransaction]
+			]
+			let data = try await postRequest(with: requestBody)
+			
+			print(String(data: data, encoding: String.Encoding.utf8))
+			
+			let response = try jsonDecoder.decode(SolanaRPCResponse<String>.self, from: data)
+			DispatchQueue.main.async { [weak self] in
```

Comment:
if I understand it correctly , update a published field doesn't need to be on the main thread,  which means you don't need to use DispatchQueue.  

Also you decorate the class with `@MainActor`  , which means it should be running on the main thread anyway

@Rockindash  is my understanding correct?

Reply:
Yes, if the class has the macro @MainActor, you don't need DispatchQueue at all. It's already running the tasks in the main thread.